src/auth/permissions/index.ts

```diff
@@ -0,0 +1,173 @@
+import * as knex from 'knex';
+import { PermissionsInitialiser, GetPermissionIds, GetRoleId } from './permissions';
+
+const getPermissionIds: GetPermissionIds = ({ client, resource, permissionLevel, access }) =>
+  client
+    .select('permission_id')
+    .table('permission')
+    .where({
+      ['permission_entity']: resource,
+      ['permission_level']: permissionLevel,
+      ['access_type']: access,
+    });
+
+const getRoleId: GetRoleId = async({ client, role }) =>
+  client
+    .select('access_role_id')
+    .table('access_role')
+    .where({ ['access_role_name']: role })
+    .then((res) => res[0].access_role_id);
+
+const permissionsInitialiser: PermissionsInitialiser = (config) => {
+  const client = knex(config);
+  return {
+    grantExisting: async ({ resource, permissionLevel, access, role }) => {
+      const roleId = await getRoleId({ client, role });
+
+      const permissionIds = await getPermissionIds({
```

Comment:
I think it's fine for this to throw instead of us performing checks on its output?

Reply:
What do you mean? I'm checking how many rows are returned to see if the correct method has been used